release-manager/files/README.md

```diff
@@ -128,8 +179,180 @@ services:
+
+To create a release bundle, the orchestration pipeline will first trigger the build of each component. Then, it will export all resources in your OpenShift namespace (`$PROJECT-$ENVIRONMENT`, here `foo-dev`) belonging to the component. By convention, this means all resources labeled with `app=$PROJECT-$COMPONENT` (e.g. `app=foo-x`). Any resources without such a label will NOT be part of the release bundle. The exported resources are stored in a `template.yml` file (an OpenShift template) located in the `openshift-exported` folder within each component repository. Further, the container image SHA of the running pod is retrieved and stored in the file `image-sha` in the same folder. Once done, the orchestration pipeline will commit the two files, tag the commit with `v1-1234-0-D` and push to the remote. After this process has been done for all repositories, the same tag is also applied to the release manager repository. At this stage, the "dev release bundle" is complete and can be installed into QA.
+
+To trigger the installation of an existing release bundle, the user needs to supply a `version` and `changeId` which has previously been used to create a release bundle. In our example, supplying `version=1`, `changeId=1234` and `environment=QA` will promote the release bundle identified by `v1-1234-0-D` to the QA environment and tag it with `v1-1234-0-Q`. Now that we have a "QA release bundle", we can promote it to PROD by supplying `version=1`, `changeId=1234` and `environment=PROD`.
+
+
+### Customizing release bundle creation
+
+As outlined above, a release bundle is essentially a state of all involved Git repositories. Each component repository contains two artifacts:
+
+- a container image SHA
+- OpenShift resource configuration (expressed in an OpenShift template)
+
+You cannot modify the image SHA (it is the result of what the component pipeline builds), but you can influence the OpenShift template. One reason to do so is that e.g. routes or `ConfigMap` values will need to differ between environments, and you need to tell the orchestration pipeline to parametrize the templates, and to supply the right values when the templates are applied in the target environment.
+
+When the orchestration pipeline exports configuration, it has no way to tell which values should actually be parameters. For example, you might have a route `x.foo-dev.dev-cluster.com` in DEV, and want this to be `x.foo-test.dev-cluster.com` in QA and `x.foo-prod.prod-cluster.com` in PROD. In the exported template, the value `x.foo-dev.dev-cluster.com` will be hardcoded. To fix this, you can create three files in the release manager repository, `dev.env`, `qa.env` and `prod.env`. These files may contain `PARAM=value` lines, like this:
```

Comment:
in the release manager repository - root folder or subfolder?

Reply:
This part I think it is already ourdated. the parameters used to customize templates per environment is done on each component, not in release manager repository anymore